commons/src/main/java/org/infinispan/commons/marshall/ProtoStreamTypeIds.java

```diff
@@ -0,0 +1,80 @@
+package org.infinispan.commons.marshall;
+
+/**
+ * TypeIds used by protostream in place of FQN message/enum names to reduce payload size.
+ * <p>
+ * ONCE SET VALUES IN THIS CLASS MUST NOT BE CHANGED AS IT WILL BREAK BACKWARDS COMPATIBILITY.
+ * <p>
+ * Values must be greater than 1000000, as this is the range reserved for internal infinispan use by the protostream
+ * project.
+ * <p>
+ * If message/enum types are no longer required, the variable should be commented instead of deleted.
+ *
+ * @author Ryan Emerson
+ * @since 10.0
+ */
+public interface ProtoStreamTypeIds {
```

Comment:
I'm not a big fan of making commons the melting pot for everything. But what else can we do?

Reply:
Yeah I agree it's a bit clunky, but if we have a Id class per module the chances of us getting overlapping Ids increases.